sensory_jamming.py

```diff
@@ -0,0 +1,342 @@
+
+    def __init__(self, popsize, boxsize, d_t, simduration):
+        self.popsize = int(popsize)
+            # integer. Size of the bat population / Number of agents 
+        self.d_t = float(d_t * m.pow(10,-3))
+            # float. Time resolution, i.e. how much time (in ms) 
+            # is expressed in 1 simulation time step.
+        self.simduration = int(simduration)
+            # integer. Total number of time steps to be run in the simulation. 
+        self.realtime = 0 
+            # time of the simulation, in seconds. Initially set at 0.
+        self.timeclock = np.empty(self.simduration, dtype = float)
+            # empty array for floats, dimensions 1*simduration. Will 
+            # contain all times in seconds for each simulation time step.
+        self.timeclock[0] = float(self.realtime) 
+            # record the first time, i.e. 0.
+        self.realduration = float(self.simduration * self.d_t)
+            # duration of the simulation in seconds.
+        self.all_ID = np.empty(self.popsize, dtype = int)
+            # empty array for integers, dimensions 1*popsize. Will 
+            # contain the identification numbers of all the agents to be run.
+        self.all_initpos = np.empty([self.popsize,2], dtype = float)
+            # empty array of intial positions for the whole population.
+        self.callsources = {}
+            # empty dictionary. Will contain the sources of each calls, 
+            # emitted by every agent throughout the simulation.
+        self.boxsize = boxsize 
+            # list of 2 integers. Area (rectangle) within which the agents can 
+            # move, defined by the lengths of edges in meters. 
+            # Has to be of the form [x,y], which is asserted below:
+        
+        assert isinstance(self.boxsize, list) and len(self.boxsize) == 2, "'boxsize' must be a list of 2 elements." 
+            # returns error if boxsize is not of the right format, i.e. [x,y]
+        
+    def Identification(self):
+        
+        for agent in range(self.popsize):
+            # for each agent in the bat population:
+            self.all_ID[int(agent)] = agent
+                # fill-in a list of all agents ID 
+                # from 0 to n = (population size - 1)
+            
+    def Positions(self):
+        
+        for ID in self.all_ID:
+            # for each agent in the bat population:
+           self.agent_initx = rd.uniform(0, self.boxsize[0])
+               # give each agent a random coordinate on the x-axis, 
+               # within the environment boundaries
+           self.agent_inity = rd.uniform(0, self.boxsize[1])
+               # give each agent a random coordinate on the y-axis, 
+               # within the environment boundaries
+           self.all_initpos[int(ID),0] = self.agent_initx
+               # store the x-coordinate in an array
+           self.all_initpos[int(ID),1] = self.agent_inity
+               # store the y-coordinate in an array
```

Comment:
@thejasvibr But the thing is, each time I change the number of agents, I have to change the matrix too. I don't think it's very efficent time-wise, and the calculations of distance that you've made earlier can be included somewhere really easily. I don't really feel like it would change that much